File: src/tui/model_picker.cpp
// src/tui/model_picker.cpp
//
// /model picker 的纯逻辑实现。把 saved_models 拆成结构化选项给
// main.cpp 的 inline overlay 渲染层 + 事件层用。
#include "model_picker.hpp"

#include <utility>

namespace acecode {

std::vector<ModelPickerOption> build_model_picker_options(
    const AppConfig& cfg, const std::string& current_name) {
    std::vector<ModelPickerOption> out;

    out.reserve(cfg.saved_models.size());
    for (const auto& e : cfg.saved_models) {
        ModelPickerOption o;
        o.name = e.name;
        o.provider = e.provider;
        o.model = e.model;
        o.is_current = (e.name == current_name);
        out.push_back(std::move(o));
    }
    return out;
}

} // namespace acecode

File: web/src/lib/modelPicker.js
// web/src/lib/modelPicker.js
// 当顶栏下拉的当前值不在 list 里(saved_models 改名/删除导致),插一条
// disabled 灰条显式提示用户。

export function isCurrentValueOrphaned(currentName, list) {
  if (!currentName) return false;
  if (!Array.isArray(list)) return false;
  return list.findIndex((m) => m && m.name === currentName) === -1;
}

export function buildOptionsWithOrphan(currentName, list) {
  if (!Array.isArray(list)) return [];
  const out = [...list];
  if (isCurrentValueOrphaned(currentName, list)) {
    out.unshift({
      name: currentName,
      provider: '?',
      model: '?',
      orphan: true,
      label: `${currentName} (已被改名/删除)`,
    });
  }
  return out;
}
